spec/classes/jira_config_spec.rb

```diff
@@ -109,15 +137,13 @@
             end
           end
 
-          context 'default params with java install, mysql, and jira > 10.3.0 and change_dbpassword=false (default) and facts that indicate a fresh install' do
+          context 'with jira > 10.3.0 and change_dbpassword=false' do
```

Comment:
Need to figure out how to stub/mock the jira::is_installed function here to be able to write test that verify the content when it returns true.

Reply:
Think I finally fixed this.
Because of Deferred() stubbing the function didn't work.
Followed https://github.com/voxpupuli/puppet-vault_lookup/pull/77/files#diff-e341d4871b1f79a6fe809f316c57bdcdd06aa8384b54c98d15b4904965407bb9 and refactored the function code into a class in the PuppetX module, and then used allow on the class method.

Also had to extend the Rspec::Puppet module with a method to clear cache in order to use allow with different return values.
Got the idea from 'option 2' here https://stackoverflow.com/questions/56852557/how-to-isolate-puppet-function-mocking-in-rspec-unit-tests

This is not mentioned in https://github.com/voxpupuli/puppet-vault_lookup/blob/master/README.md btw ..
This is not mentioned in